src/modules/gateway/sagas.ts

```diff
@@ -157,26 +166,75 @@ function* upsertPurchase(
 }
 
 function* handleStorageLoad() {
-  const pendingManaPurchase: ReturnType<typeof getPendingManaPurchase> = yield select(
-    getPendingManaPurchase
+  const pendingPurchases: ReturnType<typeof getPendingPurchases> = yield select(
+    getPendingPurchases
   )
 
-  if (pendingManaPurchase) {
-    const { network, gateway, id } = pendingManaPurchase
+  if (pendingPurchases) {
+    for (const pendingPurchase of pendingPurchases) {
+      const { network, gateway, id } = pendingPurchase
+      switch (gateway) {
+        case NetworkGatewayType.TRANSAK:
+          yield put(pollPurchaseStatusRequest(pendingPurchase))
+          break
+        case NetworkGatewayType.MOON_PAY:
+          yield put(
+            manaFiatGatewayPurchaseCompleted(
+              network,
+              gateway,
+              id,
+              MoonPayTransactionStatus.PENDING
+            )
+          )
+          break
+      }
+    }
+  }
+}
+
+function* handlePollPurchaseStatusRequest(
+  config: ManaFiatGatewaySagasConfig,
+  action: PollPurchaseStatusRequestAction
+) {
+  const { purchase } = action.payload
```

Comment:
This saga should only accept pending purchases, if by some reason it receives a completed one, it will poll forever